components/script/dom/vr.rs

```diff
@@ -128,34 +131,112 @@ impl VR {
     }
 
     pub fn handle_webvr_event(&self, event: WebVREventMsg) {
-        let WebVREventMsg::DisplayEvent(event) = event;
-        match &event {
-            &webvr::VRDisplayEvent::Connect(ref display) => {
-                let display = self.sync_display(&display);
-                display.handle_webvr_event(&event);
-                self.notify_event(&display, &event);
-            },
-            &webvr::VRDisplayEvent::Disconnect(id) => {
-                if let Some(display) = self.find_display(id) {
-                    display.handle_webvr_event(&event);
-                    self.notify_event(&display, &event);
-                }
+        match event {
+            WebVREventMsg::DisplayEvent(event) => {
+                match &event {
+                    &WebVRDisplayEvent::Connect(ref display) => {
```

Comment:
The change from `webvr::VRDisplayEvent` to `WebVRDisplayEvent` should live in its own commit to make reviewing easier.